test/framework/src/main/java/org/elasticsearch/test/AbstractSerializingTestCase.java

```diff
@@ -40,6 +45,27 @@ public final void testFromXContent() throws IOException {
             .test();
     }
 
+    /**
+     * Calls {@link ToXContent#toXContent} on many threads and verifies that they produce the same result.
```

Comment:
I think this test can use some explanation so if it fails sometimes in the future we knew which expectation this violated and understand why these expectations are reasonable.

Reply:
:+1: